cppsrc/main.cpp

```diff
@@ -109,140 +103,91 @@ Napi::FunctionReference Snowflake::constructor;
  * 
  * Function can theorotically generate 1024 unique values within a millisecond without repeating values
 */
-Napi::Value Snowflake::getUniqueIDBigInt(const Napi::CallbackInfo &info)
-{
+Napi::Value Snowflake::getUniqueIDBigInt(const Napi::CallbackInfo &info) {
     auto env = info.Env();
 
     uint64_t currentTimestamp = getCurrentTime() - this->_CUSTOM_EPOCH;
+    this->_objectId = info[0].As<Napi::Number>().Int32Value() & maxObjectId;
 
-    if (currentTimestamp == this->_lastTimestamp)
-    {
+    if (currentTimestamp == this->_lastTimestamp) {
         this->_sequence = (this->_sequence + 1) & maxSequence;
-        if (!this->_sequence)
-        {
-            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        if (!this->_sequence) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(10));
             ++currentTimestamp;
         }
-    }
-    else
-    {
+    } else {
         this->_sequence = 0;
     }
 
     this->_lastTimestamp = currentTimestamp;
 
     uint64_t id{};
-    id = currentTimestamp << (TOTAL_BITS - EPOCH_BITS);
-    id |= (this->_nodeID << (TOTAL_BITS - EPOCH_BITS - NODE_ID_BITS));
+    id = this->_objectId << (TOTAL_BITS - OBJECT_ID_BITS);
+    id |= (this->_reservedId << (TOTAL_BITS - OBJECT_ID_BITS - RESERVED_BITS)); 
+    id |= (currentTimestamp << (TOTAL_BITS - OBJECT_ID_BITS - RESERVED_BITS - EPOCH_BITS)); 
     id |= this->_sequence;
 
     return Napi::BigInt::New(env, id);
 }
 
 /**
- * Returns timestamp at which the id was generated by retreiving
- * the first 41 bits of the id, which are filled with current timestamp
+ * Returns objectId at which the id was generated by retreiving
+ * the first 11 bits of the id, which are filled with objectId
  * bits
 */
-Napi::Value Snowflake::getTimestampFromID(const Napi::CallbackInfo &info)
-{
+Napi::Value Snowflake::getObjectIDFromID(const Napi::CallbackInfo &info) {
     auto env = info.Env();
     uint64_t uniqueID{};
 
-    if (info[0].IsString())
-    {
+    if (info[0].IsString()) {
         auto first = info[0].As<Napi::String>();
-
         uniqueID = std::stoull(first.Utf8Value());
-    }
-    else if (info[0].IsBigInt())
-    {
+    } else if (info[0].IsBigInt()) {
         bool lossless = true;
         uniqueID = info[0].As<Napi::BigInt>().Int64Value(&lossless);
-    }
-    else
-    {
+    } else {
         Napi::TypeError::New(env, "BigInt or string expected").ThrowAsJavaScriptException();
     }
 
-    uint64_t timestamp = uniqueID >> (TOTAL_BITS - EPOCH_BITS);
+    uint64_t objectId = uniqueID >> (TOTAL_BITS - OBJECT_ID_BITS);
 
-    return Napi::Number::New(env, timestamp + _CUSTOM_EPOCH);
+    return Napi::Number::New(env, objectId);
 }
 
 /**
- * Returns NodeID/Machine ID at which the id was generated
+ * Returns timestamp at which the id was generated by retreiving
+ * 41 bits of the id, which are filled with current timestamp
+ * bits
```

Comment:
Same goes here and in all other occurrences:

```suggestion
 * Returns the timestamp at which the id was generated.
```